src/Mutator/AddHeaderResponseMutator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Http\Client\Common\Plugin\Cache\Mutator;
+
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Adds a header if the response came from cache.
+ *
+ * @author Iain Connor <iain.connor@priceline.com>
+ */
+class AddHeaderResponseMutator implements ResponseMutator
+{
+    /** @var string */
+    private $headerName;
+
+    /**
+     * @param string $headerName
+     */
+    public function __construct($headerName)
```

Comment:
How about adding a default value here? I guess most people will be happy with any default.

Reply:
Agreed, I'll update with a sane default.